napari/components/viewer_model.py

```diff
@@ -298,15 +292,17 @@ def reset_view(self, event=None):
             )
 
     def _new_labels(self):
-        if self.dims.ndim == 0:
-            dims = (512, 512)
-        else:
-            dims = self._calc_bbox()[1]
-            dims = [np.ceil(d).astype('int') if d > 0 else 1 for d in dims]
-            if len(dims) < 1:
-                dims = (512, 512)
-        empty_labels = np.zeros(dims, dtype=int)
-        self.add_labels(empty_labels)
+        """Create new labels layer filling full world coordinates space."""
+        extent = self.layers._extent_world
+        scale = self.layers._step_size
```

Comment:
Nice!

Reply:
Looking at it again... 😂 I wonder whether this is the right approach. If I have a points layer, scale will be 1 and it'll throw a spanner in the works here, right? I wonder if we should only look at *image* layers when determining scale...?

Reply:
@sofroniewn and I decided to punt on this one, see #1513 for tracking.

Reply:
Yup, we can revisit in next PR when we hook up `_step_size` to sliders, will be more natural to address then, and won't actually be too hard to implement :-)